docs/uml/c4_component_machine.puml

```diff
@@ -0,0 +1,37 @@
+@startuml
+!include https://raw.githubusercontent.com/plantuml-stdlib/C4-PlantUML/master/C4_Component.puml
+
+HIDE_STEREOTYPE()
+LAYOUT_TOP_DOWN()
+
+title Component diagram for DevPod with machine providers
+
+Person(user, "DevPod", "CLI or UI")
```

Comment:
I we have devpod png logo, it's possible to add image as _sprite_.
Something like

```puml
Container(devpod, "DevPod", "CLI or UI", $sprite="img:/path/to/logo.png{scale=0.7}")
```

the scale parameter is very useful to make the icon fit the diagram

Reply:
I tried the skin cloudscape-design and it looked great in the puml viewer, but unfortunately it didnt work in our website
![Screenshot from 2024-09-24 10-33-40](https://github.com/user-attachments/assets/72603695-3841-4592-902d-536840419951)

Reply:
> I tried the skin cloudscape-design and it looked great in the puml viewer, but unfortunately it didnt work in our website 

You can override background color after adding the theme, for example:

```puml
!theme cloudscape-design
skinparam BackgroundColor lightgray
```

Reply:
> Oh that would be very nice, do you know if I can define the path as a HTTP url? i.e. https://devpod.sh/images/645e37fea7fd751544ce0800_favicon1.png

This [should work](https://github.com/plantuml-stdlib/C4-PlantUML#:~:text=Additional%20$sprite%20(images)%20can%20be%20defined%20with%20following%20PlantUML%20supported%20options:), but I believe it's limited to `http`